idseq_pipeline/commands/host_filtering.py

```diff
@@ -8,6 +8,7 @@ def run(self):
         from .host_filtering_functions import *
 
         unbuffer_stdout()
+        lazy_run = not big_version_change_from_last_run(self.version, get_host_filtering_version_s3_path())
         upload_commit_sha(self.version)
 
-        run_stage1(True)
+        run_stage1(lazy_run)
```

Comment:
We have a small problem with this approach here if the pipeline crashes.   It will have uploaded a version string that suggests that any leftover files from the last run are actually from the new version, when in fact they are not.   Is there some way we can delete leftover files as soon as we decide this run shouldn't be lazy.

Reply:
Alternatively we can put the pipeline version in the path.